### PR TITLE
Fix implementation of DateFormatter and DateUtils

### PR DESCRIPTION
**Pull request by mrtnzlml**
DateFormatter now uses Intl behind the scenes and it should therefore be
 locale friendly.

### CODE REVIEW

File: app/localization/src/DateFormatter.js [hunk 1]
```diff
@@ -1,17 +1,61 @@
...
+}
+
+/**
+ * Purpose of this wrapper is to make very opinionated and restrictive
+ * formatter. Every function is localized and you should never call formatting
+ * on the raw Date object.
+ *
+ * TODO: we should probably accept custom Value Object
+ */
+function DateFormatter(rawDate: Date) {
```
Comment (tbergq) [inline app/localization/src/DateFormatter.js]: could we do `rawDate: Date = new Date()` here as well?

Comment (mrtnzlml) [reply app/localization/src/DateFormatter.js]: Why? There is not a single use-case for it and more importantly, it looks weird:

```
DateFormatter().formatToDate();
```

_"What date are you formatting here?"_ So, is it a good idea? I wanted to be explicit here...

Comment (tbergq) [reply app/localization/src/DateFormatter.js]: This is how moment works, you can do `moment().format('L')`

Comment (mrtnzlml) [reply app/localization/src/DateFormatter.js]: That's not an argument for me... 😄

Comment (tbergq) [reply app/localization/src/DateFormatter.js]: Well, I will have to live with it then 🙈

Comment (mrtnzlml) [reply app/localization/src/DateFormatter.js]: I will change it. As I said there is not a single use-case of `DateFormatter(new Date())` in our codebase... 🙂

Comment (tbergq) [reply app/localization/src/DateFormatter.js]: Ok, OCD calming.. 😄 Ping me when it is done

Comment (mrtnzlml) [reply app/localization/src/DateFormatter.js]: Your OCD should be calmed down now. I also introduced new `formatToBirthday` in order to format birthdays. Not sure about the name but it's good enough at this point...

**Comment by mrtnzlml** [comment]
@tbergq 🚀

**Comment by mrtnzlml** [comment]
This PR is very approved... 😄

**Comment by tbergq** [comment]
I'm a click-jacker 🙈 🌮 🎉

**Pull request merged**